chronos/ratelimit.nim

```diff
@@ -0,0 +1,109 @@
+#               Chronos Rate Limiter
+#            (c) Copyright 2022-Present
+#         Status Research & Development GmbH
+#
+#              Licensed under either of
+#  Apache License, version 2.0, (LICENSE-APACHEv2)
+#              MIT license (LICENSE-MIT)
+
+{.push raises: [Defect].}
+
+import ../chronos
+import timer
+
+export timer
+
+type
+  RateCounter* = object
+    budget: int64
+    budgetCap: int64
+    lastUpdate: Moment
+
+  RateCost* = distinct int64
+
+proc update(rc: var RateCounter) =
+  let
+    currentTime = Moment.now()
+    timeDelta = nanoseconds(currentTime - rc.lastUpdate)
+
+  rc.lastUpdate = currentTime
+  rc.budget = min(rc.budgetCap, rc.budget + timeDelta)
+
+proc operationsPerSecond*(ops: float): RateCost =
+  ## Create a RateCost from an allowed number of 
+  ## operations per second
+  const budgetPerSecond = 1.seconds.nanoseconds.float
+
+  RateCost(budgetPerSecond / ops)
```

Comment:
What the reason to perform conversion to `float` perform `float` division and convert result back to integer? I think number of operations SHOULD not be float because i dont see how count could be rational number.